decode.go

```diff
@@ -328,7 +349,7 @@ func newDecoder() *decoder {
 	d := &decoder{
 		stringMapType:  stringMapType,
 		generalMapType: generalMapType,
-		uniqueKeys:     true,
```

Comment:
This is enforced by the spec. I'm not sure we want to allow tuning it anymore. There's just no good reason to accept broken documents.

Reply:
Do you think we just need to remove control over it? In proposed design it's just a matter or removal `UniqueKeys` method from `DecodeOptions` and it will be not changeable. Though it's strange that it's used in code 3 times if this feature is not needed - may be it would be worth to remove all corresponding `if d.options.uniqueKeys` statements.